Backend/BackendAPI/Controllers/BikesController.cs

```diff
@@ -0,0 +1,105 @@
+﻿using BackendAPI.Models;
+using BackendAPI.Models.DTOFactories;
+using BackendAPI.Repository.Interfaces;
+using ClassLibrary.DTO;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace BackendAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize]
+    [ApiController]
+    public class BikesController : ControllerBase
+    {
+        private int GetRequestingUserID => int.Parse(
+                User.FindFirstValue(ClaimTypes.NameIdentifier));
+
+        private IRentalRepository rentalRepository;
+        private IBikeRepository bikeRepository;
+        private IUserRepository userRepository;
+
+        public BikesController(IRentalRepository rentalRepo, IBikeRepository bikeRepo, IUserRepository userRepo)
+        {
+            rentalRepository = rentalRepo;
+            bikeRepository = bikeRepo;
+            userRepository = userRepo;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        [HttpGet("rented")]
+        public ActionResult RentedGet()
+        {
+            int userId = GetRequestingUserID;
+
+            BikeDTO[] rentedBikes = rentalRepository.FindActiveRentals(userId)
+                .Select(r => BikeDTOFactory.CreateBikeDTO(r.Bike, r.User)).ToArray();
```

Comment:
.ToList() jest szybsze niż .ToArray() przy tego typu rzeczach z tego co kiedyś czytałem, zresztą ja bym najlepiej zostawił tego Selecta i po nim już nic nie zmieniał. Wtedy rentedBikes będzie miało typ IEnumerable, ale na spokojnie można zwrócić taki typ (json to i tak przerobi sobie na tablicę)